pkg/kubelet/checkpointmanager/checkpoint_manager.go

```diff
@@ -79,8 +80,8 @@ func (manager *impl) GetCheckpoint(checkpointKey string, checkpoint Checkpoint)
 	defer manager.mutex.Unlock()
 	blob, err := manager.store.Read(checkpointKey)
 	if err != nil {
-		if err == utilstore.ErrKeyNotFound {
-			return errors.ErrCheckpointNotFound
+		if errors.Is(err, utilstore.ErrKeyNotFound) {
```

Comment:
is there a linter for this? Without linter this seems to fall under https://www.kubernetes.dev/docs/guide/pull-requests/#trivial-edits

it is also hard to confirm that this will not change functionality. What if some error was wrapped before and `==` didn't return `true` in some cases and this PR will change this.

Reply:
Thank you for your suggestion. I have checked the two implementations of the `Read`method, and there are no special cases; this PR will not change any logic.

Reply:
I think the point is what is the benefit for being correct vs assuming a risk of depending on a third party library that may change some behavior and break our compoent (one that clusters depend completely on them)

Reply:
I think the point is what is the benefit for being correct vs assuming a risk of depending on a third party library that may change some behavior and break our compoent

Reply:
I reckon it's all good to bring in the errors package right from the get-go in the implementation. When it comes to making this change, there's a bit of risk involved on one side, and honestly, I don't see any major benefits from this tweak right off the bat. That's just my two cents on it, so if I'm off the mark, feel free to point it out. Thanks

Reply:
> is there a linter for this? Without linter this seems to fall under https://www.kubernetes.dev/docs/guide/pull-requests/#trivial-edits
> 
> it is also hard to confirm that this will not change functionality. What if some error was wrapped before and `==` didn't return `true` in some cases and this PR will change this.

If an error has been wrapped, shouldn't we use errors.Is to determine it, as == might not yield the expected result?

Reply:
I assume that the concerns of above other reviews is about risk. 

In the point view of code cleaning up, I really support to replace all `==` to `errors.Is`. However, considering this scenario: Maybe, just maybe, some logic are dependent on wrapped error to make the `==` always return `false`. Changing to `errors.Is` without digging everyplace deeply may lead some unexpected behavior, so we have to be really careful to do so.

Reply:
> I assume that the concerns of above other reviews is about risk.
> 
> In the point view of code cleaning up, I really support to replace all `==` to `errors.Is`. However, considering this scenario: Maybe, just maybe, some logic are dependent on wrapped error to make the `==` always return `false`. Changing to `errors.Is` without digging everyplace deeply may lead some unexpected behavior. so to accomplish this migration, we have to be really really careful.

https://github.com/kubernetes/kubernetes/blob/026c55e40de835464e769bad65c8a19940b61459/pkg/kubelet/util/store/store.go#L41-L54

Thank you for your reminder. For my PR, this change is safe. `Read`must return ErrKeyNotFound if key is not found.

Reply:
> I assume that the concerns of above other reviews is about risk.
> 
> In the point view of code cleaning up, I really support to replace all `==` to `errors.Is`. However, considering this scenario: Maybe, just maybe, some logic are dependent on wrapped error to make the `==` always return `false`. Changing to `errors.Is` without digging everyplace deeply may lead some unexpected behavior. so to accomplish this migration, we have to be really really careful.

I understand he's not just talking about this method, but all the changes involved.

Reply:
I completely agree with the idea you want to implement, but perhaps we should consider it more thoroughly. Just as @xuzhenglun  mentioned, we need to proceed with caution. Additionally, if we want to make this change, maybe we should make this commit more meaningful. Can we add this restriction to the linter to ensure consistent styling?

Reply:
/close

based on this discussion and the fact that it fails under the trivial edit policy